ui/src/App.js

```diff
@@ -4,66 +4,65 @@ import * as Stomp from 'stomp-websocket';
 import axios from 'axios';
 import './App.css';
 
-var stompClient = null;
-var serverMessage = null;
+let stompClient = null;
 
 class App extends Component {
   state = {
-    response: null,
+    apiResponse: null,
+    serverMessage: null,
+    connected: false,
+    name: '',
```

Comment:
Made a number of changes to this component to make it more React-ified. Won't go into to a load of detail, but you can read here if you want more info on what React `state` is, and how it's treated specially within a React component: 
https://reactjs.org/docs/state-and-lifecycle.html
https://reactjs.org/docs/faq-state.html